amivapi/auth/utils.py

```diff
@@ -8,6 +8,29 @@
 from os import urandom
 
 
+def init_secret(app):
+    """Get secret key from database or create it. Put in config.
+
+    The secret key is stored in the database to ensure consistency.
+    If no secret is in the database yet, create it.
+
+    The database collection holding this key is called `config`
+    """
+    with app.app_context():  # App context needed for db connection
+        key = 'TOKEN_SECRET'
+        db = app.data.driver.db['config']
+        result = db.find_one({key: {'$exists': True, '$nin': [None, '']}})
+
+        if result is not None:
+            secret = result[key]
+        else:
+            # Create secret and save in db
+            secret = b64encode(urandom(32)).decode('utf_8')
```

Comment:
Not every sequence is valid utf-8, so this might fail somehow.

Reply:
ah I missed the b64encode

Reply:
It's first base64 encoded, which can always be utf-8 decoded.